python/src/dataset.rs

```diff
@@ -719,15 +720,30 @@ impl Dataset {
                         ivf_params.precomputed_partitons_file = Some(f.to_string());
                     };
 
-                    if let Some(o) = kwargs.get_item("shuffle_partition_batches")? {
-                        ivf_params.shuffle_partition_batches =
-                            PyAny::downcast::<PyInt>(o)?.extract()?;
-                    };
-
-                    if let Some(o) = kwargs.get_item("shuffle_partition_concurrency")? {
-                        ivf_params.shuffle_partition_concurrency =
-                            PyAny::downcast::<PyInt>(o)?.extract()?;
-                    };
+                    match (
+                        kwargs.get_item("precomputed_shuffle_buffers")?,
+                        kwargs.get_item("precomputed_shuffle_buffers_path")?
+                    ) {
+                        (Some(l), Some(p)) => {
+                            let path = Path::parse(p.to_string()).map_err(|e| {
+                                PyValueError::new_err(format!(
+                                    "Failed to parse precomputed_shuffle_buffers_path: {}",
+                                    e
+                                ))
+                            })?;
+                            let list = PyAny::downcast::<PyList>(l)?
```

Comment:
I think this will require `precomputed_shuffle_buffers` to be a list (and not, for example, any arbitrary iterable of strings).  Let's make sure to document this fact well if we make this externally documented.

Reply:
oh yeah these are not exposed in the docs. users are not expected to use these APIs.